server/services/bookingService.ts

```diff
@@ -50,28 +54,51 @@ export default class BookingService {
     const bookingClient = this.bookingClientFactory(callConfig)
     const bookings = await bookingClient.allBookingsForPremisesId(premisesId)
 
+    const lostBedClient = this.lostBedClientFactory(callConfig)
+    const lostBeds = await (
+      await lostBedClient.allLostBedsForPremisesId(premisesId)
+    ).filter(lostBed => lostBed.status === 'active')
+
     const bedId = room.beds[0].id
 
-    return bookings
-      .filter(booking => booking?.bed.id === bedId)
+    const allReservations = [...bookings, ...lostBeds]
```

Comment:
Feel free to reject this, but I'm wondering if we can lose the need for `isBooking` - it makes me slightly uneasy that we know what's a void and what's a booking, we then lose that information by putting them in an array togehter, and we then derive it again by being lucky that they don't share a particular field.

We could do something like -

```
const foo1 = bookings.map(b => {sortingValue: v.arrivalDate, rows: [...]}
const foo2 = voids.map(v => {sortingValue: v.arrivalDate, rows: [...]}

return [...foo1, foo2].sort(...).map(bookingOrVoid => bookingOrVoid.rows)
```

We then don't rely on being able to distinguish between voids and bookings in a roundabout way, and we don't need to do `isBooking` for each row

Reply:
yes, good idea. Have implemented